File: 74. Search a 2D Matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        for row in matrix:
            if target >= row[0] and target <= row[len(row)-1]:
                return True if target in row else False
        return False

File: 74. Search a 2D Matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        for (int i = 0; i < matrix.size(); i++)
        {
            if (target >= matrix[i][0] && target <= matrix[i][matrix[i].size() - 1])
                return find(matrix[i].begin(), matrix[i].end(), target) != matrix[i].end() ? true : false;
        }
        return false;
    }
};
